it/splunk/src/main/java/com/google/cloud/teleport/it/splunk/DefaultSplunkResourceManager.java

```diff
@@ -0,0 +1,375 @@
+  private static final String DEFAULT_SPLUNK_USERNAME = "admin";
+
+  private final ServiceArgs loginArgs;
+  private final int hecPort;
+  private final String hecScheme;
+  private final String hecToken;
+
+  @SuppressWarnings("resource")
+  private DefaultSplunkResourceManager(DefaultSplunkResourceManager.Builder builder) {
+    this(
+        new SplunkContainer(
+                DockerImageName.parse(builder.containerImageName)
+                    .withTag(builder.containerImageTag))
+            .withSplunkdSslDisabled(),
+        builder);
+  }
+
+  @VisibleForTesting
+  DefaultSplunkResourceManager(
+      SplunkContainer container, DefaultSplunkResourceManager.Builder builder) {
+    super(setup(container, builder), builder);
+
+    // Validate builder args
+    if (builder.useStaticContainer) {
+      if (builder.hecPort < 0 || builder.splunkdPort < 0) {
+        throw new SplunkResourceManagerException(
+            "This manager was configured to use a static resource, but the hecPort and splunkdPort were not properly set.");
+      }
+    }
+
+    String username = DEFAULT_SPLUNK_USERNAME;
+    if (builder.useStaticContainer && builder.username != null) {
+      username = builder.username;
+    }
+    hecPort =
+        builder.useStaticContainer ? builder.hecPort : container.getMappedPort(builder.hecPort);
+    int splunkdPort =
+        builder.useStaticContainer
+            ? builder.splunkdPort
+            : container.getMappedPort(builder.splunkdPort);
+
+    // TODO - add support for https scheme
+    String splunkdScheme = "http";
+    hecScheme = "http";
+
+    hecToken = builder.hecToken;
+
+    loginArgs = new ServiceArgs();
+    loginArgs.setPort(splunkdPort);
+    loginArgs.setHost(this.getHost());
+    loginArgs.setScheme(splunkdScheme);
+    loginArgs.setUsername(username);
+    loginArgs.setPassword(builder.password);
+  }
+
+  /**
+   * Helper method for injecting config information from the builder into the given SplunkContainer.
+   *
+   * @param container The SplunkContainer before config info is injected.
+   * @param builder The DefaultSplunkResourceManager.Builder to extract config info from.
+   * @return The SplunkContainer with all config info injected.
+   */
+  private static SplunkContainer setup(SplunkContainer container, Builder builder) {
+    builder.hecPort = builder.hecPort < 0 ? DEFAULT_SPLUNK_HEC_INTERNAL_PORT : builder.hecPort;
+    builder.splunkdPort =
+        builder.splunkdPort < 0 ? DEFAULT_SPLUNKD_INTERNAL_PORT : builder.splunkdPort;
+    builder.hecToken = builder.hecToken == null ? generateHecToken() : builder.hecToken;
+    builder.password = builder.password == null ? generateSplunkPassword() : builder.password;
+    return container
+        .withDefaultsFile(
+            Transferable.of(
+                String.format(
+                    "splunk:\n"
+                        + "  hec:\n"
+                        + "    enable: True\n"
+                        + "    ssl: %b\n"
```

Comment:
Case disagreement

```
enable: True
ssl: false
```


Assuming not an issue here?

Reply:
Case does not matter, but I will switch True to true for consistency